client/core/core.go

```diff
@@ -608,6 +608,19 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 		previousStatus := trade.metaData.Status
 		newStatus := order.OrderStatus(srvOrderStatus.Status)
 		trade.metaData.Status = newStatus
+		// If there is an associated cancel order, and we are revising the
+		// status of the targeted order, we can infer the cancel order is done.
+		if trade.cancel != nil {
+			cid := trade.cancel.ID()
+			err := trade.db.UpdateOrderStatus(cid, order.OrderStatusRevoked)
```

Comment:
This bit, for updating the deleted cancel order status, is not done in `trackedTrade.deleteStaleCancelOrder()`, perhaps adding that there will prevent such cancel orders from being loaded from db on restart.